db/db_encryption_test.cc

```diff
@@ -116,6 +116,55 @@ TEST_F(DBEncryptionTest, ReadEmptyFile) {
   ASSERT_TRUE(data.empty());
 }
 
+TEST_F(DBEncryptionTest, NotSupportedGetFileSize) {
+  // Encrypted envs do not support GetFileSize.
+  // Validate ReadFooterFromFile allowing not supported GetFileSize file system.
```

Comment:
The abstraction level of this test is odd to me. If we are "testing the test" to confirm that EncryptedEnv covers the scenario of FSRandomAccessFile::GetFileSize returning NotSupported, that can be a much smaller and more directed test. If we are trying to test RocksDB's behavior with such a FileSystem, that should be covered by all the integration tests (in unit test binaries) running under CI with ENCRYPTED_ENV=1.

There is certainly a time and a place for unit testing components, but ReadFooterFromFile seems so dependent on integrated behaviors that unit testing it (vs. integration testing) just creates friction to code evolution.